.github/workflows/cookstyle.yml

```diff
@@ -10,18 +10,8 @@ jobs:
     - name: Check out code
       uses: actions/checkout@v3
     - name: Setup ruby
-      uses: actions/setup-ruby@v1.1.3
-    - name: Cache gems
-      uses: actions/cache@v3
+      uses: ruby/setup-ruby@v1
       with:
-        path: vendor/bundle
-        key: bundle-${{ runner.OS }}-${{ hashFiles('Gemfile.lock') }}
-        restore-keys: |
-          bundle-${{ runner.OS }}-
-    - name: Install gems
-      run: |
-        gem install bundler
-        bundle config set deployment true
-        bundle install --jobs 4 --retry 3
+        bundler-cache: true
```

Comment:
I know this is what they tell you to do but I'm not convinced it correctly encodes the OS version... The closest I can see is that I think it uses `common.getVirtualEnvironmentName()` as part of the cache key.

The use of `runner.OS` in our keys was very deliberate because we had issues when `ubuntu-latest` changed from one version to the next with needing to make sure the cache was invalidated.

Reply:
Actually I think that does work, given the code at https://github.com/ruby/setup-ruby/blob/master/common.js#L104.